django/db/backends/base/operations.py

```diff
@@ -714,42 +714,51 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             "This backend does not support %s subtraction." % internal_type
         )
 
-    def window_frame_start(self, start):
-        if isinstance(start, int):
-            if start < 0:
-                return "%d %s" % (abs(start), self.PRECEDING)
-            elif start == 0:
+    def window_frame_value(self, value, unbound):
+        if isinstance(value, int):
+            if value < 0:
+                return "%d %s" % (abs(value), self.PRECEDING)
+            elif value == 0:
                 return self.CURRENT_ROW
-        elif start is None:
-            return self.UNBOUNDED_PRECEDING
-        raise ValueError(
-            "start argument must be a negative integer, zero, or None, but got '%s'."
-            % start
-        )
-
-    def window_frame_end(self, end):
-        if isinstance(end, int):
-            if end == 0:
-                return self.CURRENT_ROW
-            elif end > 0:
-                return "%d %s" % (end, self.FOLLOWING)
-        elif end is None:
-            return self.UNBOUNDED_FOLLOWING
-        raise ValueError(
-            "end argument must be a positive integer, zero, or None, but got '%s'."
-            % end
-        )
+            else:
+                return "%d %s" % (value, self.FOLLOWING)
+        elif value is None:
+            return unbound
```

Comment:
We raise `ValueError` before calling this method so we can reach this code only with `None`
```suggestion
        return unbound
```

Reply:
I wonder if it's worth passing `unbound`, maybe:
```python
start_ = self.window_frame_value(start) or self.UNBOUNDED_PRECEDING
end_ = self.window_frame_value(end) or self.UNBOUNDED_FOLLOWING
```